streamer/pulp/streamer/server.py

```diff
@@ -174,84 +113,182 @@ def _handle_get(self, request):
+                        self._on_succeeded(entry, request, last_report)
+                        return
+                    except (DownloadFailed, DoesNotExist, PluginNotFound):
+                        # try another
+                        continue
+                # Failed
+                self._on_all_failed(request)
             except Exception:
-                logger.exception(_('An unexpected error occurred while handling the request.'))
+                logger.exception(_('An unexpected error occurred.'))
                 request.setResponseCode(INTERNAL_SERVER_ERROR)
 
-    def _download(self, catalog_entry, request, responder):
-        """
-        Build a nectar downloader and download the content from the catalog entry.
-        The download is performed by the alternate content container, so it is possible
-        to use the streamer in conjunction with alternate content sources.
-
-        :param catalog_entry:   The catalog entry to download.
-        :type  catalog_entry:   pulp.server.db.model.LazyCatalogEntry
-        :param request:         The client content request.
-        :type  request:         twisted.web.server.Request
-        :param responder:       The file-like object that nectar should write to.
-        :type  responder:       Responder
-        """
-        # Configure the primary downloader for alternate content sources
-        plugin_importer, config, db_importer = repo_controller.get_importer_by_id(
-            catalog_entry.importer_id)
-        # There is an unfortunate mess of configuration classes and attributes, and
-        # multiple "models" floating around. The MongoEngine class that corresponds
-        # to the database entry only contains the repository config. The ``config``
-        # variable above contains the repository configuration _and_ the plugin-wide
-        # configuration, so here we override the db_importer.config because it doesn't
-        # have the whole config. In the future the importer object should seemlessly
-        # load and apply the plugin-wide configuration.
-        db_importer.config = config.flatten()
-        primary_downloader = plugin_importer.get_downloader_for_db_importer(
-            db_importer, catalog_entry.url, working_dir='/tmp')
-        pulp_request = request.getHeader(PULP_STREAM_REQUEST_HEADER)
-        listener = StreamerListener(request, self.config, catalog_entry, pulp_request)
-        primary_downloader.session = self.session
-        primary_downloader.event_listener = listener
-
-        # Build the alternate content source download request
-        unit_model = plugins_api.get_unit_model_by_id(catalog_entry.unit_type_id)
-        qs = unit_model.objects.filter(id=catalog_entry.unit_id).only(*unit_model.unit_key_fields)
+    def _on_succeeded(self, entry, request, report):
+        """
+        The download succeeded.
+
+        :param entry: A catalog entry.
+        :type  entry: LazyCatalogEntry
```

Comment:
Since `LazyCatalogEntry` has been imported, it does not need to be fully qualified.